lib/Reader/reader.cpp

```diff
@@ -7158,6 +7161,12 @@ void ReaderBase::readBytesForFlowGraphNodeHelper(
       getCallInfo(&ResolvedToken, nullptr /*constraint*/,
                   CORINFO_CALLINFO_LDFTN, &CallInfo);
 
+      // ReadyToRun has delegate constructor optimization for non-virtual
+      // function pointers resolved at compile time only for now.
+      if ((Flags & CORJIT_FLG_READYTORUN) && (CallInfo.kind != CORINFO_CALL)) {
```

Comment:
It's the latter. I re-phrased the comments to make it more clear:

`// Currently ReadyToRun has delegate constructor optimization only for`
 `// non-virtual function pointers resolved at compile time.`